src/utilities/graphqlConnectionFactory.ts

```diff
@@ -36,3 +38,154 @@ export function graphqlConnectionFactory<T>(): InterfaceConnection<T> {
+  return args.cursor ? args.limit + 2 : args.limit + 1;
+};
+
+// Generates the sortingObject that can be passed in the .sort() method
+export const getSortingObject = (
+  args: ConnectionArguments,
+  sortingObject: { [key: string]: number }
+) => {
+  // We assume that the resolver would always be written with respect to the sorting that needs to be implemented for forward pagination
+  if (args.direction === "FORWARD") return sortingObject;
+
+  // If we are paginating backwards, then we must reverse the order of all fields that are being sorted by.
+  for (const [key, value] of Object.entries(sortingObject)) {
+    sortingObject[key] = value * -1;
+  }
+
+  return sortingObject;
+};
+
+// Generates the sorting arguments for filterQuery that can be passed into the .find() method
+export function getFilterQuery(args: ConnectionArguments) {
+  if (args.cursor) {
+    if (args.direction === "FORWARD") return { _id: { $gte: args.cursor } };
+    else return { _id: { $lte: args.cursor } };
+  }
+
+  return null;
+}
+
+/*
+This is a function which generates a GraphQL cursor pagination object based on the requirements of the client.
+
+The function takes the following arguments: 
+
+A. TYPE PARAMETERS
+
+1. T: Refers the type of the node that the connection and it's edges would refer. 
+Example values include `Interface_User`, `Interface_Organization`, `Interface_Event`, `Interface_Post`.
+2. U: Regers to the type of interface that is implemented by the model that you want to query. 
+For example, if you want to query the TagUser Model, then you would send Interface_UserTag for this type parameter
+
+B. DATA PARAMETERS
+
+1. args: These are the tranformed arguments that were orginally passed in the request.
+2. allFetchedObjects: Refers to objects that were fetched from the database thorugh a query.
+3. getNodeFromResult: Describes a transformation function that given an object of type U, would convert it to the desired object of type T. This would mostly include mapping to some specific field of the fetched object.
+
+The function returns a promise which would resolve to the desired connection object (of the type InterfaceConnection<T>).
+*/
+export function generateConnectionObject<
+  T extends { _id: Types.ObjectId },
+  U extends { _id: Types.ObjectId }
+>(
+  args: ConnectionArguments,
+  allFetchedObjects: U[] | null,
+  getNodeFromResult: GetNodeFromResultFnType<T, U>
+): InterfaceConnectionResult<T> {
+  // Initialize the connection object
+  const connectionObject = graphqlConnectionFactory<T>();
+
+  // Return the default object if the recieved list is empty
+  if (!allFetchedObjects || allFetchedObjects.length === 0)
+    return {
+      connectionData: connectionObject,
+      connectionErrors: null,
+    };
+
+  // Handling the case when the cursor is provided
+  if (args.cursor) {
+    // Validate the cursor
+    if (allFetchedObjects[0]._id.toString() !== args.cursor.toString()) {
```

Comment:
Same here, I think mongoose's `.equals()` method to compare objectIds is better and cleaner.